nautobot/dcim/filters.py

```diff
@@ -333,11 +344,93 @@ class LocationFilterSet(NautobotFilterSet, StatusModelFilterSetMixin, TenancyFil
         field_name="location_type__content_types",
         choices=FeatureQuery("locations").get_choices,
     )
+    has_circuit_terminations = RelatedMembershipBooleanFilter(
+        field_name="circuit_terminations",
+        label="Has circuit terminations",
+    )
+    devices = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=Device.objects.all(),
+        to_field_name="name",
+        label="Devices (name or ID)",
+    )
+    has_devices = RelatedMembershipBooleanFilter(
+        field_name="devices",
+        label="Has devices",
+    )
+    # The reverse relation here is misnamed as `powerpanel`, but fixing it would be a breaking API change.
```

Comment:
it actually is `powerpanels` currently for Location; we should look at fixing this to `power_panels` as a part of #2520.